### PR TITLE
Fix user listing error

### PR DESCRIPTION
**Pull request by hypery2k**
FIXED #JENKINS-18884

See [JENKINS-18884](https://issues.jenkins-ci.org/browse/JENKINS-18884).

And People page of All view also displays all users registered in Jenkins.

They do no permission check.

This patch does followings:

Adds permission check to jobs to extract users (now requires Read permission).
People page of All view no longer display all users registered in Jenkins. It works same as those of other views:
There is no reason for People page of All page to work different from those of other views.
Displaying all users in Jenkins should be handled by security realms (like HudsonPrivateSecurityRealm.ManageUserLinks). 
This is another approach of #1094 .

Readdition of #1102
### Proposed changelog entries

* View of users requiring overall read permission

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@oleg-nenashev

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
Will try to review, but likely it will happen after Sep 25

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
The newly introduced test fails on both Linux and Windows

```
java.lang.AssertionError
at org.junit.Assert.fail(Assert.java:86)
at org.junit.Assert.assertTrue(Assert.java:41)
at org.junit.Assert.assertFalse(Assert.java:64)
at org.junit.Assert.assertFalse(Assert.java:74)
at hudson.model.AsynchPeopleTest.testProjectPermission(AsynchPeopleTest.java:101)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
at org.jvnet.hudson.test.JenkinsRule$1.evaluate(JenkinsRule.java:548)
at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at java.lang.Thread.run(Thread.java:748)
```

**Comment by daniel-beck** [comment]
Please also configure your IDE to not add pointless whitespace changes, they make a PR difficult to review.

**Comment by hypery2k** [comment]
sorry, happened by accident. Should I create a new PR ?

**Comment by oleg-nenashev** [comment]
@hypery2k you can replace the changes in this PR if it is prefereable. We can squash changes during the merge so it won't affect the main repo history

**Comment by daniel-beck** [comment]
Still tons of unnecessary whitespace changes. 20+ lines before line 700 in `View.java` are changed unnecessarily.

**Comment by oleg-nenashev** [comment]
I wish GitHub had an easy chunk revert proposal functionality

File: core/src/main/java/jenkins/util/ProgressiveRendering.java [hunk 1]
```diff
@@ -203,6 +203,13 @@ private static void setCurrentRequest(RequestImpl request) {
         }
     }
 
+    /**
+     * @return whether the computation has finished.
+     */
+    public boolean isFinished() {
```
Comment (oleg-nenashev) [inline core/src/main/java/jenkins/util/ProgressiveRendering.java]: If this is test-only, it should be restricted

File: core/src/main/java/hudson/model/View.java [hunk 2]
```diff
@@ -865,27 +864,12 @@ public AsynchPeople(View parent) {
...
-                            modified.add(u);
-                        }
-                    }
-                }
-            }
+        }
+
+        // for testing purpose
+        @Restricted(NoExternalUse.class)
+            /*package*/ Set<User> getModified() {
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/View.java]:
```suggestion
     /*package*/ Set<User> getModified() {
```

**Review by oleg-nenashev — APPROVED** [review]
Apart from the formatting mess, the pull request looks good to me. It cannot be backported in the current state, but IMHO it's fine as security hardening

File: core/src/main/java/jenkins/util/ProgressiveRendering.java (hunk 1 above)
Comment (hypery2k) [reply core/src/main/java/jenkins/util/ProgressiveRendering.java]: fixed ;)

**Review by daniel-beck — CHANGES_REQUESTED** [review]
At a minimum, the whitespace changes need to be reverted and the history cleaned up so there's no possible change of merging unsquashed.

Between backports to LTS and the way we develop security fixes, this is likely to result in conflicts for no good reason.

**Comment by hypery2k** [comment]
created a new clean PR, HTH

**Pull request closed without merging**